File: AlgoPersonalStudy/src/backjoon/silver/BJ_1764.java
package backjoon.silver;

import java.io.*;
import java.util.*;

public class BJ_1764 {
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        HashSet<String> nameSet = new HashSet<>();
        for(int n=0;n<N;n++){
            String name = br.readLine();
            nameSet.add(name);
        }
        List<String> list = new ArrayList<>();
        StringBuilder sb = new StringBuilder();
        for(int m=0;m<M;m++){
            String name = br.readLine();
            if(nameSet.contains(name)){
                list.add(name);
            }
        }
        int pqSize = list.size();
        Collections.sort(list);
        for(String name : list){
            sb.append(name).append("\n");
        }
        System.out.println(pqSize);
        System.out.println(sb.toString());
//        PriorityQueue<String> pq = new PriorityQueue<>();
//        StringBuilder sb = new StringBuilder();
//        for(int m=0;m<M;m++){
//            String name = br.readLine();
//            if(nameSet.contains(name)){
//                pq.add(name);
//            }
//        }
//        int pqSize = pq.size();
//        for(String name : pq){
//            sb.append(name).append("\n");
//        }
//        System.out.println(pqSize);
//        System.out.println(sb.toString());
    }
}

File: AlgoPersonalStudy/src/_kotlin/backjoon/silver/BJ_1764.kt
package _kotlin.backjoon.silver

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

class BJ_1764 {
    fun solution(){
        val br = BufferedReader(InputStreamReader(System.`in`))
        val st = StringTokenizer(br.readLine())
        val N = st.nextToken().toInt()
        val M = st.nextToken().toInt()
        val nameSet = HashSet<String>()
        for (n in 0 until N) {
            val name = br.readLine()
            nameSet.add(name)
        }
        val list: MutableList<String> = ArrayList()
        val sb = StringBuilder()
        for (m in 0 until M) {
            val name = br.readLine()
            if (nameSet.contains(name)) {
                list.add(name)
            }
        }
        val pqSize = list.size
        list.sort()
        for (name in list) {
            sb.append(name).append("\n")
        }
        println(pqSize)
        println(sb.toString())
    }
}
fun main(){
    val solutionClass = BJ_1764()
    solutionClass.solution()
}